desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,818 @@
 """Methods for computing bounce integrals (singular or otherwise)."""
 
+import numpy as np
 from interpax import CubicHermiteSpline, PPoly
+from numpy.fft import rfft2
 from orthax.legendre import leggauss
 
-from desc.backend import jnp
+from desc.backend import dct, jnp
+from desc.integrals.basis import FourierChebyshevSeries, PiecewiseChebyshevSeries
 from desc.integrals.bounce_utils import (
     _bounce_quadrature,
     _check_bounce_points,
+    _check_interp,
     _set_default_plot_kwargs,
     bounce_points,
+    get_alpha,
     get_pitch_inv_quad,
     interp_to_argmin,
     plot_ppoly,
 )
-from desc.integrals.interp_utils import polyder_vec
+from desc.integrals.interp_utils import (
+    cheb_from_dct,
+    cheb_pts,
+    interp_rfft2,
+    irfft2_non_uniform,
+    polyder_vec,
+)
 from desc.integrals.quad_utils import (
     automorphism_sin,
+    bijection_from_disc,
     get_quadrature,
     grad_automorphism_sin,
+    grad_bijection_from_disc,
 )
 from desc.io import IOAble
-from desc.utils import errorif, setdefault, warnif
+from desc.utils import (
+    atleast_nd,
+    check_posint,
+    errorif,
+    flatten_matrix,
+    setdefault,
+    warnif,
+)
+
+
+def _transform_to_desc(grid, f, is_reshaped=False):
+    """Transform to DESC spectral domain.
+
+    Parameters
+    ----------
+    grid : Grid
+        Tensor-product grid in (θ, ζ) with uniformly spaced nodes in
+        (2π × 2π) poloidal and toroidal coordinates.
```

Comment:
should zeta be 2pi/NFP?

Reply:
I have to think more about how the corresponding potential change to jnp.fftfreq in interp_utils. It might be trivial, but periodicity etc. is lost in different coordinates so I have to be careful